### ISSUE TITLE
One share of SMB server is not indexed/scanned (completely)

### ISSUE THREAD

**Issue by bobvandevijver** [owncloud/core#29067]
### Steps to reproduce
On a single SMB share from one server, the files are not indexed correctly. Sometimes I have a fraction of the folders available (9 out of 36), but most of the time the folder is empty or not even accessible from the Owncloud interface.
The behavior stays even after re-adding the share configuration. Note that I have most shared configured twice: once with guest access for every used, and once with password access for myself.

A scan from the command line simply skips the directory (filtered log of `sudo -u www-data php occ files:scan -vvv bobvandevijver`): 
```
Starting scan for user 1 out of 1 (bobvandevijver)                                                                                                                                                                                        
        Folder /bobvandevijver/                                                                                                                                                                                                           
        Folder /bobvandevijver/files                                                                                                                                                                                                      
        File   /bobvandevijver/files/woonplaats-scraper.rar                                                                                                                                                                               
        Folder /bobvandevijver/files/[BOBV] Series/                                                                                                                                                                                       
        Folder /bobvandevijver/files/[BOBV] Software/                                                                                                                                                                                     
        File   /bobvandevijver/files/[BOBV] Software/XFI_SupportPack_2_5.exe   
        Folder /bobvandevijver/files/[BOBV] Software/WinRAR
```
As you can see, the Series directory is skipped while all other shares with the same configuration are fully indexed.

The mount configuration:
```
+----------+------------------+--------------+------------------------------+------------------------------------------------------------------------------------------------------------------------------+----------------------+
| Mount ID | Mount Point      | Storage      | Authentication Type          | Configuration                                                                                                                | Options              |
+----------+------------------+--------------+------------------------------+------------------------------------------------------------------------------------------------------------------------------+----------------------+
| 8        | /[BOBV] Private  | SMB / CIFS   | Gebruikersnaam en wachtwoord | host: "localhost", share: "Private", root: "", domain: "", user: "bobvandevijver", password: "***"                           | enable_sharing: true |
| 9        | /[BOBV] Uploads  | SMB / CIFS   | Gebruikersnaam en wachtwoord | host: "localhost", share: "Uploads", root: "", domain: "", user: "bobvandevijver", password: "***"                           | enable_sharing: true |
| 10       | /[BOBV] Study    | SMB / CIFS   | Gebruikersnaam en wachtwoord | host: "localhost", share: "Study", root: "", domain: "", user: "bobvandevijver", password: "***"                             | enable_sharing: true |
| 11       | /[BOBV] Software | SMB / CIFS   | Gebruikersnaam en wachtwoord | host: "localhost", share: "Software", root: "", domain: "", user: "bobvandevijver", password: "***"                          | enable_sharing: true |
| 12       | /[BOBV] Series   | SMB / CIFS   | Gebruikersnaam en wachtwoord | host: "localhost", share: "Series", root: "", domain: "", user: "bobvandevijver", password: "***"                            | enable_sharing: true |
| 13       | /[BOBV] Music    | SMB / CIFS   | Gebruikersnaam en wachtwoord | host: "localhost", share: "Music", root: "", domain: "", user: "bobvandevijver", password: "***"                             | enable_sharing: true |
| 14       | /[BOBV] Movies   | SMB / CIFS   | Gebruikersnaam en wachtwoord | host: "localhost", share: "Movies", root: "", domain: "", user: "bobvandevijver", password: "***"                            | enable_sharing: true |
| 15       | /Google Drive    | Google Drive | OAuth2                       | configured: "true", client_id: "476911...nt.com", client_secret: "***", token: "***"                                         | enable_sharing: true |
| 17       | /Dropbox         | Dropbox      | OAuth1                       | configured: "true", app_key: "***", app_secret: "***", token: "***", token_secret: "***"                                     |                      |
+----------+------------------+--------------+------------------------------+------------------------------------------------------------------------------------------------------------------------------+----------------------+
```

### Expected behaviour
All enabled Samba shares should be visible in the interface, and all should have all files listed and accessible.

### Actual behaviour
For one of the shares (Series) the files are not visible/accessible, when it is possible to open the folder. It might also be the case that the "Storage is not available" message is shown, but that message disappears for no apparent reason.

All other shares on the same server with the same configuration work as expected.

### Server configuration
**Operating system**: Debian 9 (Stretch)

**Web server:** Apache/2.4.25 (2.4.25-3+deb9u3)

**Database:** MariaDb 2.4.25-3+deb9u3

**PHP version:** 7.0.19-1, with the pecl extension smbclient installed.

**ownCloud version:** 10.0.3.3 (10.0.3-1.1 from stable repository)

**Updated from an older ownCloud or fresh install:** Updated from 10.0.2, but this problem was already present there

**Where did you install ownCloud from:** Debian repositories

**Signing status (ownCloud 9.0 and above):** 
https://gist.github.com/bobvandevijver/b8d19a4bc56de65457b2cccb4f8d36b1

**The content of config/config.php:**
https://gist.github.com/bobvandevijver/3c927e663a92503abbf62f601a1775fd

**List of activated apps:**

```
Enabled:                                                                                                                                                                                                                                  
  - activity: 2.3.6                                                                                                                                                                                                                       
  - comments: 0.3.0                                                                                                                                                                                                                       
  - configreport: 0.1.1                                                                                                                                                                                                                   
  - dav: 0.3.0                                                                                                                                                                                                                            
  - federatedfilesharing: 0.3.1                                                                                                                                                                                                           
  - federation: 0.1.0                                                                                                                                                                                                                     
  - files: 1.5.1                                                                                                                                                                                                                          
  - files_external: 0.7.1                                                                                                                                                                                                                 
  - files_pdfviewer: 0.8.2                                                                                                                                                                                                                
  - files_sharing: 0.10.1                                                                                                                                                                                                                 
  - files_texteditor: 2.2.1                                                                                                                                                                                                               
  - files_trashbin: 0.9.1                                                                                                                                                                                                                 
  - files_versions: 1.3.0                                                                                                                                                                                                                 
  - files_videoplayer: 0.9.8                                                                                                                                                                                                              
  - firstrunwizard: 1.1                                                                                                                                                                                                                   
  - gallery: 16.0.2                                                                                                                                                                                                                       
  - notifications: 0.3.1                                                                                                                                                                                                                  
  - provisioning_api: 0.5.0                                                                                                                                                                                                               
  - systemtags: 0.3.0                                                                                                                                                                                                                     
  - templateeditor: 0.1                                                                                                                                                                                                                   
  - updatenotification: 0.2.1                                                                                                                                                                                                             
Disabled:                                                                                                                                                                                                                                 
  - encryption                                                                                                                                                                                                                            
  - external                                                                                                                                                                                                                              
  - files_antivirus                                                                                                                                                                                                                       
  - market                                                                                                                                                                                                                                
  - theme-example                                                                                                                                                                                                                         
  - user_external
```

**Are you using external storage, if yes which one:** smb/google drive/dropbox

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** N/A

**Operating system:** N/A

### Logs
#### Web server error log
There is no relevant information in the log.

#### ownCloud log (data/owncloud.log)
There is no relevant information in the log.

#### Browser log
N/A

**Comment by PVince81** [comment]
Are there maybe files with special characters in that one share that isn't fully scanned ? Does the user configured in OC have permissions to scan deeper ? Maybe the subfolders in questions are not accessible for the SMB user ?

**Comment by bobvandevijver** [comment]
The share is actually configured with both a full-access user-account and an guest account: both have the same behavior. The share is actually an open share that can be accessed by any user, similar to another share that does work in Owncloud. The affected share is completely accessible by the guest user when using another smbclient, however, I did not check all subfolders.ince81 

I did not yet check for specific characters in the file names. However, I just did and found a script with an symbolic link to another share and tried to remove that first: that worked like a charm and directly solved my issues!

However, it should be noted that this symbolic link did work in other clients. So I'm not sure whether this is an Owncloud bug or a bug in de smbclient library. I will leave this issue open for you to decide.
